core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/DDiagramContents.java

```diff
@@ -295,8 +297,15 @@ public Collection<DDiagramElement> getDiagramElements(EObject target, DiagramEle
 
     ArrayList<DDiagramElement> result = new ArrayList<>();
     for (DDiagramElement view : getMapDiagramElements().get(target)) {
-      if (mapping != null && mapping.equals(view.getDiagramElementMapping())) {
+      DiagramElementMapping diagramElementMapping = view.getDiagramElementMapping();
+      if (mapping != null && mapping.equals(diagramElementMapping)) {

```

Comment:
maybe the DiagramService.isMapping is ok no? rather than those new methods

Reply:
isMapping's test is instanceof instead of equals.
It is also on two DDiagramElements (instead of DiagramElementMapping).

Was the comment on the wrong code section?

Reply:
Hmm no, DiagramService.isMapping  `public boolean isMapping(DDiagramElement element, DiagramElementMapping mapping)`  is calling sirius helper isInstanceOf which is testing equals..

Reply:
mis à jour.

Reply:
can you push it again ? 
i probably rebased your previous change and lose your change.

Reply:
Done.

Reply:
Am I suppose to mark it resolved or should I let this decision to you @pdulth ? (asking both for now and future tickets)

Reply:
Mainly 
- make sure that each different changes have a corresponding github issue
- your PR commits can refer to different github issues at once (no need one PR per issue) as long as those are separated in different commits
- if needed put the PR in draft until review to avoid a commiter review (sometimes you need just a build not a review)
- don't bother with copyright update dates in file header
- don't bother with issue status, they are updated all at once while we release (but make sure your commits references the issue #id at beginning of message to ease it) (here we have direct access on which issue have been fixed https://github.com/eclipse/capella/commits/master)